packages/jellyfish-api-core/__tests__/category/loan/setLoanToken.test.ts

```diff
@@ -0,0 +1,258 @@
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import BigNumber from 'bignumber.js'
+import { Testing } from '@defichain/jellyfish-testing'
+import { GenesisKeys } from '@defichain/testcontainers'
+
+describe('Loan', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should setLoanToken', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token1',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token1',
+      name: 'Token1',
+      priceFeedId
+    })
+    expect(typeof loanTokenId).toStrictEqual('string')
+    expect(loanTokenId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    expect(data).toStrictEqual({
+      [loanTokenId]: {
+        token: {
+          1: {
+            symbol: 'Token1',
+            symbolKey: 'Token1',
+            name: 'Token1',
+            decimal: 8,
+            limit: 0,
+            mintable: false,
+            tradeable: true,
+            isDAT: true,
+            isLPS: false,
+            finalized: false,
+            isLoanToken: true,
+            minted: 0,
+            creationTx: loanTokenId,
+            creationHeight: await testing.container.getBlockCount(),
+            destructionTx: '0000000000000000000000000000000000000000000000000000000000000000',
+            destructionHeight: -1,
+            collateralAddress: expect.any(String)
+          }
+        },
+        priceFeedId,
+        interest: 0
+      }
+    })
+  })
+
+  it('should setLoanToken if symbol is more than 8 letters', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'x'.repeat(8),
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'x'.repeat(9), // 9 letters
+      name: 'Token2',
+      priceFeedId
+    })
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].symbol).toStrictEqual('x'.repeat(8)) // Only remain the first 8 letters
```

Comment:
wondering why they allowed it instead of rejecting if `symbol` > 8 🤔

Reply:
Agreed, it is confusing this way

Reply:
I think we just let it be at this moment. They borrow the idea from AppointOracle. (See the image below)

<img width="806" alt="Screenshot 2021-09-09 at 5 26 30 PM" src="https://user-images.githubusercontent.com/7101806/132660490-eaf9a096-8d5c-490e-924e-8f40caef082f.png">

Reply:
Have asked Diego about this. let's see. But for now, this is the c++ side implementation.